include/ezgl/canvas.hpp

```diff
@@ -34,6 +36,9 @@ namespace ezgl {
 /**** Functions in this class are for ezgl internal use; application code doesn't need to call them ****/
 
 class renderer;
```

Comment:
Should comment what these 3 functions are for. 

Also, I think you're going to pass these surfaces into the create_context routine, and then draw normally with the resulting context for it to go to a file.  What closes the file -- destroying the context?  I think there should be a higher-level comment somewhere describing the steps in drawing to a file, and that drawing (hopefully) occurs normally except for the creation of the surface and context and perhaps the deleting of the context (or something else?) to close the file.